Source/Charts/Data/Implementations/Standard/ChartDataEntry.swift

```diff
@@ -114,3 +114,12 @@ extension ChartDataEntry/*: Equatable*/ {
             && x == object.x
     }
 }
+
+extension ChartDataEntry: Comparable {
```

Comment:
I don't see how sort by `x`, then by `y` isn't the obvious way to sort two dimensional data. If there is indeed no obvious way to sort, then we need to get rid of binary search altogether. At best it is slow, at worst it doesn't work.

Reply:
Also, all of the code that relies on sort assumes things are sorted by x (which currently there is no guarantee of this), and then, if needed, loops through all equal `y` values to find the one we're looking for. It is faster to just sort the `y` values as well and use binary search.

Reply:
For example it could be sorted by the distance from zero. Or by `y` then by `x` or just by `x` or just by `y`. That depends on the data represented by points on chart, how it's natural to compare two of them. I don't say there shouldn't be any sorting or comparison. I just say, that IMHO if function is added to some type, it is supposed to work correct for any value of this type so probably if function will work as expected only for array controlled (and sorted) by chart, it seems to be better to add this function to chart not to all random access collections (and even better it would be if it is impossible to use an array without this sorting for chart data)